File: src/Easy/rangeSumOfBST/rangeSumOfBST.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} L
 * @param {number} R
 * @return {number}
 */
var rangeSumBST = function(root, L, R) {
  if (!root.val) return 0;
  let total = 0;

  const shell = (root) => {
    if (!root) return null;

    if (root.val <= R && root.val >= L) {
      total += root.val;
    }
  };

  shell(root.left);
  shell(root.right);

  return total;
};

module.exports = rangeSumBST;

File: src/Easy/rangeSumOfBST/rangeSumOfBST.java
/**
 * Definition for a binary tree node. public class TreeNode { int val; TreeNode
 * left; TreeNode right; TreeNode(int x) { val = x; } }
 */
class Solution {
  int sum = 0;

  public int rangeSumBST(TreeNode root, int L, int R) {
    helper(root, L, R);
    return sum;
  }

  public void helper(TreeNode root, int L, int R) {
    if (root == null) {
      return;
    }

    if (root.val >= L && root.val <= R) {
      sum += root.val;
      helper(root.left, L, R);
      helper(root.right, L, R);
    } else if (root.val < L) {
      helper(root.right, L, R);
    } else {
      helper(root.left, L, R);
    }
  }

}
